soufi/finders/centos.py

```diff
@@ -101,17 +95,17 @@ def get_source_repos(self, subdirs: Iterable[str], optimal_repos: bool):
         to run the generator over when re-walking the list of repos is minimal.
         """
         for dir in self._get_dirs():
-            for subdir in subdirs:
+            for subdir in self.repos:
```

Comment:
The unmolested default value for `self.repos` is `None`, which is non-iterable; how is this not shattering to pieces in tests?

Reply:
Because:
```
        if not repos or optimal_repos is True:
            repos = DEFAULT_SEARCH
```